### ISSUE TITLE
Using RMove to Generate Recommendations?

### ISSUE THREAD

**Issue by Abhiram98** [wangsiqidahaoren/Rmove-method#2]
Thank you for your valuable contribution to the field of refactoring! Your ICSME'22 paper provided useful insights into applying machine learning techniques for software engineering tasks. I am interested in running RMove on my own project to auto-generate refactoring recommendations. For a given class, I'd like to understand what methods need to move and where? 

Unfortunately, I was unable to achieve this task by looking at your source code. Could you please help me understand how to obtain refactoring recommendations from RMove. In particular, I have two questions:

1. How can I vectorize new methods and classes, to run RMove on them? 
Is it similar to the way you created the training data? Do we pass the method and class bodies to an embedding and get the vectors? Would it be possible to provide some details about this?

2. How can we trigger RMove on a class to generate move-method recommendations? 
After reading the paper and looking at the source code, it wasn't clear how we could use a classifier (which generates true/false labels given a <method, class> pair) to generate recommendations. 

Here is my understanding of how we can generate recommendations using RMove: For a given method, we use RMove exhaustively against every class in the project to predict true/false labels. Then, we interpret every positive prediction as a recommendation made by RMove. I.e., if RMove predicts true for <methodA, classB>, it is a recommendation to move methodA to classB. We then repeat this process for every method in a given class to generate a complete list of recommendations for a given class. 
What do you think of this? Is this how you would generate recommendations?

**Comment by wangsiqidahaoren** [comment]
Hi there,
Thank you for contact.

You can train the code and save the classify model(LR, CNN...) parameter, and then do the same thing like training data even on a single item. When RMove predicts true for <methodA, classB>, it is a recommendation to move methodA to classB.

You should notice that classifier cross projects may not have a very good result. We really recommend that you train specific classifiers for your projects

**Comment by Abhiram98** [comment]
Thanks for clarifying! 
Can you help me understand what scripts I should trigger to generate the training data? Is the training data the same for all models?
